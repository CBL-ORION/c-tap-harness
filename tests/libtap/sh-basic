#! /bin/sh
#
# Calls basic libtap functions for testing the shell library.
#
# Written by Russ Allbery <rra@stanford.edu>
# Copyright 2008, 2009 Board of Trustees, Leland Stanford Jr. University
#
# See LICENSE for licensing terms.

# Load the test library.
. "$SOURCE/../tap/libtap.sh"

# Plan test.
plan 25

# ok tests.
ok true
ok false
ok [ 1 -eq 1 ]
ok [ 2 -eq 1 ]

# skip tests.
skip no idea why

# ok_block tests.
ok_block 1 false
ok_block 2 true
ok_block 3 [ 1 -ne 1 ]

# skip_block tests.
skip_block 1 some reason
skip_block 2 some other reason

# ok_program tests.
ok_program 0 '' /bin/true
ok_program 0 '' sh -c 'exit 1'
ok_program 1 '' sh -c 'exit 1'
ok_program 0 'test is successful' echo test is successful
ok_program 0 'test fails' sh -c 'echo test fails; exit 1'
ok_program 0 'test fails' echo test failed
ok_program 1 'yay' sh -c 'echo yay; exit 1'
ok_program 10 'yay' sh -c 'echo yay; exit 10'
ok_program 10 'yay' sh -c 'echo yay; exit 9'
ok_program 1 'test' echo test
ok_program 1 'test' sh -c 'echo yay; exit 1'
