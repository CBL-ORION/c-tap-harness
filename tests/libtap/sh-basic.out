1..25
ok 1
not ok 2
ok 3
not ok 4
ok 5 # skip no idea why
not ok 6
ok 7
ok 8
not ok 9
not ok 10
not ok 11
ok 12 # skip some reason
ok 13 # skip some other reason
ok 14 # skip some other reason
ok 15
#  saw: (1) 
#  not: (0) 
not ok 16
ok 17
ok 18
#  saw: (1) test fails
#  not: (0) test fails
not ok 19
#  saw: (0) test failed
#  not: (0) test fails
not ok 20
ok 21
ok 22
#  saw: (9) yay
#  not: (10) yay
not ok 23
#  saw: (0) test
#  not: (1) test
not ok 24
#  saw: (1) yay
#  not: (1) test
not ok 25
